### PR TITLE
set fog of war configuration

### PR DESCRIPTION
**Pull request by vadymkutsenko**
Added posibility to configure FogOfWar for selected game modes
Just add desirable mode to config at section at section`"plugin": "FogOfWar",` to `allowedForGameModes` field

For example:
`"allowedForGameModes": ["RAAS"]`

~~disclaimer: changes have not yet been tested at locall environment~~

checked locally - OK

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -209,6 +209,7 @@
       "plugin": "FogOfWar",
       "enabled": false,
       "mode": 1,
+      "allowedForGameModes": [],
```
Comment (Svdovych) [inline config.json]: Set all possible gamemodes as default value, for reference

File: squad-server/plugins/fog-of-war.js [hunk 2]
```diff
@@ -39,8 +44,17 @@ export default class FogOfWar extends BasePlugin {
   }
 
   async onNewGame() {
-    setTimeout(() => {
-      this.server.rcon.setFogOfWar(this.options.mode);
+    setTimeout(async () => {
+      if (this.options.mode === 1 && this.options.allowedForGameModes.length > 0) {
+        const currentMap = await this.server.rcon.getCurrentMap();
+        this.options.mode = this.options.allowedForGameModes.some((gameMode) =>
```
Comment (Svdovych) [inline squad-server/plugins/fog-of-war.js]: Won't this overwrite the configured value for mode and disable fog of war for all modes once it hits one mode that's not allowed?

File: config.json (hunk 1 above)
Comment (vadymkutsenko) [reply config.json]: added

File: squad-server/plugins/fog-of-war.js (hunk 2 above)
Comment (vadymkutsenko) [reply squad-server/plugins/fog-of-war.js]: thanks, good point
 changed to a more secure implementation

**Pull request merged**